### PR TITLE
Migrating hot restart operations to use IMDG client

### PR DESCRIPTION
**Pull request by erosb**
Related MC issue: https://github.com/hazelcast/management-center/issues/2224

Related client-protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/278

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/ClientAddClusterViewListenerCodec.java [hunk 1]
```diff
@@ -37,45 +37,50 @@
 /**
  * TODO DOC
  */
-@Generated("5d1772412b87cda68e87d8ac98ba3bec")
+@Generated("31932e989ca413ed287b11a2e8f57174")
```
Comment (puzpuzpuz) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/ClientAddClusterViewListenerCodec.java]: Please merge your client protocol PR with `master` and run generator once again. This codec is not related with hot restart operations and it has changed because this repo and the client protocol one were out of sync. It was fixed by https://github.com/hazelcast/hazelcast-client-protocol/pull/277

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupMessageTask.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    private final Node node;
+
+    HotRestartBackupOperation(Node node) {
+        this.node = node;
+    }
+
+    @Override
+    public void run()
+            throws Exception {
+        node.getNodeExtension().getHotRestartService().backup();
```
Comment (emre-aydin) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupMessageTask.java]: Why do we need to have this in a dedicated operation? AFAICT it's enough to call this method on a single member, which would then act as a coordinator and trigger hot backup on all members.

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupInterruptMessageTask.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    private final Node node;
+
+    public HotRestartBackupInterruptOperation(Node node) {
+        this.node = node;
+    }
+
+    @Override
+    public void run()
+            throws Exception {
+        node.getNodeExtension().getHotRestartService().interruptBackupTask();
```
Comment (emre-aydin) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupInterruptMessageTask.java]: Same with trigger hot backup. This method already causes interruption on all members. It's enough to execute it on a single coordinator member.

Comment (erosb) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupInterruptMessageTask.java]: Fixed, thanks.

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupMessageTask.java (hunk 2 above)
Comment (erosb) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/HotRestartBackupMessageTask.java]: Fixed, thanks.

**Comment by puzpuzpuz** [comment]
Closing this one in favor of #16262
